File: LeetCode/2373. Largest Local Values in a Matrix/Solution.java
/**
 * Runtime: 2 ms
 * Memory Usage: 42.4 MB 
 */
class Solution {
    public int maxPoolingKernel(int[][] grid, int x, int y) {   
        int mx = 0;
        for (int i = x; i < x + 3; ++ i) {
            for (int j = y; j < y + 3; ++ j) {
                mx = Math.max(mx, grid[i][j]);
            }
        }
        return mx;
    }
    public int[][] largestLocal(int[][] grid) {
        int n = grid.length;
        int[][] ans = new int[n - 2][n - 2];
        for (int i = 0; i < n - 2; ++ i) {
            for (int j = 0; j < n - 2; ++ j) {
                ans[i][j] = maxPoolingKernel(grid, i, j);
            }
        }
        return ans;
    }
}

File: LeetCode/2373. Largest Local Values in a Matrix/solution.py
"""
Runtime: 112 ms
Memory Usage: 15.4 MB
"""
class Solution:
    def largestLocal(self, grid: List[List[int]]) -> List[List[int]]:
        def max_pooling_kernel(x, y):
            mx = 0
            for i in range(x, x + 3):
                for j in range(y, y + 3):
                    mx = max(mx, grid[i][j])
            return mx
        n = len(grid)
        ans = [[0] * (n - 2) for _ in range(n - 2)]
        for i in range(n - 2):
            for j in range(n - 2):
                ans[i][j] = max_pooling_kernel(i, j)
        return ans
